### ISSUE TITLE
missing OAUTH_REFRESH_TOKEN

### ISSUE THREAD

**Issue by Dusan308a** [andreafabrizi/Dropbox-Uploader#525]
I have followed the procedure for generating new App_key, App_secret and Access_code within Dropbox after ticking the tickmarks in the Permissions section. I have copied these on the first run of the dropbox_uploader.sh and confirmed the entry since all three of them looked OK when ECHOed back to the console. However, the line in the "dropbox_uploader" file with OAUTH_REFRESH_TOKEN remains empty after this. Could someone suggest what I did wrong? Thanks!
Dusan

**Comment by gmcinalli** [comment]
@Dusan308a I can confirm this, just write your refresh token directly into the config file and you're good to go.

**Comment by Dusan308a** [comment]
Hi,
Thanks for the fast response. However, I still have problems, Dropbox will
not let me in.
I long have a Dropbox account, and I have just these days created a
connected App as specified at your site, and I can see it in Dropbox
personal account. I have obtained the AppKey, AppSecret and Access token
(mine is significantly longer than the one shown at your internet page and
has 137 charasters starting with "sl.A"). I have tick-marked options for
this connected app on the "permissions" page of the Dropbox setup.I have a
folder on Dropbox to use for this purpose and it is shared (I think not
needed really).
I run the 'dropbox_uploader' for the first time, and it asks me on the
above keyword, which I enter. It asks me to confirm the entered keys, and I
confirm with "y". When I go and view the file '.dropbox_uploader' the
fourth line with "OAUTH_REFRESH_TOKEN=" remains empty; nothing is written
after the "=" and when I run the ' dropbox_uploader ' again I get the
message "Error loading data from ...." and "It is recommended to run ...
unlink".
If I manually copy the Access token into this lline of the
".dropbox_uploader", and run the './dropbox_uploader.sh list /', the
response changes to "> Listing "/"...FAILED" and "Some error occured.
....", and 'the du_resp' file reads "Error in call to API function
"files/list_folder": Invalid authorization value in HTTP header
"Authorization": "Bearer".  Expecting "Bearer <oauth2-access-token>"."
I am sorry to bother you again since I am sure I made a silly beginner's
mistake somewhere. However, I would be grateful for help.
Dušan

**Comment by lpadula** [comment]
I have exactly the same problem, and it's driving me crazy! Have you find a solution?

**Comment by Dusan308a** [comment]
Hi
No, I do not have the solution. I am new with Raspberry Pi, Linux and JS
and was kind of hoping for some help. I have read a low of stuff these past
days and I suspect something was changed from Dropbox side, and that Andrea
suggestions do not work any more; please correct me if I made a wrong
assumption. The Dropbox page Andrea mentioned has changed, and suggests the
use of something called DBX platform. Also, the page to select the kind of
access differs from the one Andrea showed,, and the length of the key-code
so I am in doubt. I have read the Dropbox page
https://www.dropbox.com/developers, and have selected *here* to get to
https://www.dropbox.com/developers/documentation?_tk=pilot_lp&_ad=altcta1&_camp=docs
with detailed explanation of the drivers to install from Dropbox site to
access the Dropbox from different applications, ald langueges, including
JS. I will try those instructions instead, but expect sirious problems due
to my lack of knowledge.
Good luck!
Dusan

**Comment by telemaxx** [comment]
i have just tested that and can confirm this.
after .dropbox_uploader was created i checked it and the fourth line was OK. 
than I entered the "list" command and got the error. now the fourth line was empty.

so I removed ".dropbox_uploader"
and started the script and entered the api keys stuff again. 

before using the list command i copied the .dropbox_uploader file to have a backup.

now I tried directly uploading a file and it worked. not sure why.

**Comment by Dusan308a** [comment]
Hi
I gave up the Dropbox-Uploader since I was unable to make it work on my
Raspberry Pi zero W. After dozens of (different) trys I always got the same
result, that is the empty fourth line. I must be doing something wrong, but
cannot figure out what, and yes, it is frustrating.
I did, however, managed to upload/download/list files from my dropbox
account using "rclone" as decribed at "
https://www.thedigitalpictureframe.com/how-to-finally-access-dropbox-on-your-raspberry-pi-using-rclone/".
Everything worked on first attempt, and I am going to proceed with this
software. Thanks for all suggestions!
Regards, Dusan

**Comment by telemaxx** [comment]
rclone seems to be interesting. as dropbox-uploader is not really maintained anymore,  it is good to an alternative.

**Comment by tjluoma** [comment]
FYI - As much as I like/enjoy/respect dropbox_uploader. I have to say that `rclone` seems to have this token situation handled much better. I was able to get `rclone` setup very quickly, and it's handling the token issue seamlessly.

**Comment by diggidre** [comment]
> i have just tested that and can confirm this.
> after .dropbox_uploader was created i checked it and the fourth line was OK.
> than I entered the "list" command and got the error. now the fourth line was empty.
> 
> so I removed ".dropbox_uploader"
> and started the script and entered the api keys stuff again.
> 
> before using the list command i copied the .dropbox_uploader file to have a backup.
> 
> now I tried directly uploading a file and it worked. not sure why.

I can confirm. At first clean install the .dropbox_uploader file is partly empty. After deleting it and running the dropbox_uploader.sh again the .dropbox_uploader config was filled correctly.
During second time setup I generated a new Refresh-Token... Im not sure if this does matter.
Then after second setup the file uploading was fixed.

**Comment by barbalex** [comment]
Yep, rclone seems great. Check https://rclone.org/remote_setup/ for setup in a headless box and https://rclone.org/docs/#copying-single-files to copy single files
